### PR TITLE
refactor(core): remove unused or testing-only utilities from DomAdapters

### PR DESCRIPTION
**Pull request by kara**
`DomAdapter` is brought into every Angular application by default, but it's pretty large in size. This PR attempts to clean out the parts of `DomAdapter` that are:

- completely unused at the moment
- used only by tests run in a browser environment (in which case the utils are moved to platform-browser)
- jasmine matchers used only by tests run in a browser environment

For these cases, there's no point including the code into production bundles. The moved code will be tree-shaken away. This seems to save about 3-4KB.

None of the parts of `DomAdapter` that are actually used at runtime should be affected.

Note: This PR only tackles half of `DomAdapter` because the PR started to get big. There will be follow-up PRs.

### CODE REVIEW

**Review by pkozlowski-opensource — APPROVED** [review]
@kara this is totally awesome cleanup!!! LVGTM for the runtime parts I understand :-)

File: aio/scripts/_payload-limits.json [hunk 1]
```diff
@@ -28,7 +28,7 @@
       "uncompressed": {
         "runtime-es5": 2932,
         "runtime-es2015": 2938,
-        "main-es5": 560811,
+        "main-es5": 554933,
```
Comment (gkalpak) [inline aio/scripts/_payload-limits.json]: How come only ES5 is affected?

Comment (kara) [reply aio/scripts/_payload-limits.json]: That's a good question 🤔 I'll look into it

Comment (kara) [reply aio/scripts/_payload-limits.json]: Ah, turns out it's in both. It just happened to not hit the threshold to throw. Updating...

**Review by kyliau — APPROVED** [review]
LGTM for language service

**Review by mhevery — APPROVED** [review]

**Comment by kara** [comment]
Caretaker note: will need patch CL

[VE presubmit](https://test.corp.google.com/ui#id=OCL:264976604:BASE:265117669:1566590618570:bbfa33ca)

[Ivy presubmit](https://test.corp.google.com/ui#id=OCL:264976604:BASE:265110688:1566588553205:eda3a403)

**Comment by kara** [comment]
merge-assistance: global approval

**Pull request closed without merging**